sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOIT.java

```diff
@@ -70,159 +65,166 @@
+            .apply("Hash row contents", Combine.globally(new HashingFn()).withoutDefaults());
+
+    PAssert.that(consolidatedHashcode)
+        .containsInAnyOrder(TestRow.getExpectedHashForRowCount(options.getNumberOfRecords()));
+
+    pipelineRead.run().waitUntilFinish();
+  }
 
-    private final String tableName;
-    private final Cluster cluster;
+  private static Cluster getCluster(CassandraIOITOptions options) {
+    return Cluster.builder()
+        .addContactPoints(options.getCassandraHost().toArray(new String[0]))
+        .withPort(options.getCassandraPort())
+        .build();
+  }
 
-    CassandraMatcher(Cluster cluster, String tableName) {
-      this.cluster = cluster;
-      this.tableName = tableName;
-    }
-
-    @Override
-    protected boolean matchesSafely(PipelineResult pipelineResult) {
-      pipelineResult.waitUntilFinish();
-      Session session = cluster.connect();
-      ResultSet result = session.execute("select id,name from " + CassandraTestDataSet.KEYSPACE
-          + "." + tableName);
-      List<Row> rows = result.all();
-      if (rows.size() != 1000) {
-        return false;
-      }
-      for (Row row : rows) {
-        if (!row.getString("name").matches("Name.*")) {
-          return false;
-        }
-      }
-      return true;
+  private static void createTable(CassandraIOITOptions options, String keyspace, String tableName) {
+    try (Cluster cluster = getCluster(options);
+        Session session = cluster.connect()) {
+      LOG.info("Create {} keyspace if not exists", keyspace);
+      session.execute(
+          "CREATE KEYSPACE IF NOT EXISTS "
+              + KEYSPACE
+              + " WITH REPLICATION = "
+              + "{'class':'SimpleStrategy', 'replication_factor':3};");
+
+      session.execute("USE " + keyspace);
+
+      LOG.info("Create {} table if not exists", tableName);
+      session.execute(
+          "CREATE TABLE IF NOT EXISTS "
+              + tableName
+              + "(id bigint, name text, PRIMARY "
+              + "KEY(id))");
     }
+  }
 
-    @Override
-    public void describeTo(Description description) {
-      description.appendText("Expected Cassandra record pattern is (Name.*)");
+  private static void dropTable(CassandraIOITOptions options, String keyspace, String table) {
+    try (Cluster cluster = getCluster(options);
+        Session session = cluster.connect()) {
+      session.execute("DROP KEYSPACE IF EXISTS " + keyspace);
+      session.execute("DROP TABLE IF EXISTS " + keyspace + "." + table);
     }
   }
 
-  /**
-   * Simple Cassandra entity representing a scientist. Used for read test.
-   */
-  @Table(name = CassandraTestDataSet.TABLE_READ_NAME, keyspace = CassandraTestDataSet.KEYSPACE)
-  static class Scientist implements Serializable {
+  /** Simple Cassandra entity representing a scientist. Used for read test. */
+  @Table(name = TABLE, keyspace = KEYSPACE)
+  static final class Scientist implements Serializable {
```

Comment:
forgot it in the review: I think this can be private (compiles, runs too).

Reply:
ok